### PR TITLE
[hive] repair table support sync all properties to hms 

### PR DESCRIPTION
**Pull request by herefree**
### Purpose

Linked issue: close #xxx
Follow up #4038

### CODE REVIEW

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java [hunk 1]
```diff
@@ -819,6 +819,20 @@ private void updateHmsTable(Table table, Identifier identifier, TableSchema sche
     private void updateHmsTablePars(Table table, TableSchema schema) {
         if (syncAllProperties()) {
             table.getParameters().putAll(schema.options());
+            if (!schema.primaryKeys().isEmpty()) {
```
Comment (JingsongLi) [inline paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java:822]: Please extract a method to reuse.

**Comment by JingsongLi** [comment]
> Just curious，in real scene Paimon should not support alter pk，partitionkey？ @herefree

Yes

**Comment by herefree** [comment]
> Just curious，in real scene Paimon should not support alter pk，partitionkey？ @herefree
Yes, if you have some history paimon table which was not syn pk or partitionkey to hive meta, you can use repair table.It also need to add some code to support it, just like this.
image

**Comment by xuzifu666** [comment]
> > Just curious，in real scene Paimon should not support alter pk，partitionkey？ @herefree
> 
> Yes, if you have some history paimon table which was not syn pk or partitionkey to hive meta, you can use repair table.It also need to add some code to support it, just like this. image

Get，but the title maybe not fitable，alter primary-key can take a look at SparkSchemaEvolutionITCase##testSetAndRemoveOption which can not support.

**Comment by herefree** [comment]
> > > Just curious，in real scene Paimon should not support alter pk，partitionkey？ @herefree
> > 
> > 
> > Yes, if you have some history paimon table which was not syn pk or partitionkey to hive meta, you can use repair table.It also need to add some code to support it, just like this. image
> 
> Get，but the title maybe not fitable，alter primary-key can take a look at SparkSchemaEvolutionITCase##testSetAndRemoveOption which can not support.

Yes,you are right,i'll change the title.

**Comment by JingsongLi** [comment]
+1

**Pull request merged**